linker.go

```diff
@@ -301,9 +301,10 @@ func fixupKfuncs(insns asm.Instructions) (_ handles, err error) {
 
 fixups:
 	// only load the kernel spec if we found at least one kfunc call
-	kernelSpec, err := btf.LoadKernelSpec()
-	if err != nil {
-		return nil, err
+	if kernelSpec == nil {
+		if kernelSpec, err = btf.LoadKernelSpec(); err != nil {
+			return nil, err
```

Comment:
@lmb I think adding some context to the error string would be more helpful to both devs and users:

```suggestion
			return nil, fmt.Errorf("kfuncs require the kernel to be built with BTF: %w", err)
```